components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/internal/BasicAuthenticatorServiceComponentTestCase.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.application.authenticator.basicauth.internal;
+
+import org.apache.commons.logging.Log;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.osgi.service.component.ComponentContext;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.lang.reflect.Field;
+
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doAnswer;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class BasicAuthenticatorServiceComponentTestCase {
+
+    private String debugMsg;
+    private RealmService mockRealmService;
+    private BasicAuthenticatorServiceComponent basicAuthenticatorServiceComponent;
+    private ComponentContext mockComponentContext;
+    private Log mockedLog;
+
+    @BeforeTest
+    public void setup() {
+
+        basicAuthenticatorServiceComponent = new BasicAuthenticatorServiceComponent();
+    }
+
+    @Test
+    public void setRealmTestCase() throws NoSuchFieldException, IllegalAccessException {
+
+        mockedLog = mock(Log.class);
+        mockRealmService = mock(RealmService.class);
+        enableDebugLogs(mockedLog);
+
+        doAnswer(new Answer<Object>() {
```

Comment:
use can use a doNothing().when(mockedLog).debug(anyString())
to achieve the same.